izanami-frontend/src/pages/users.tsx

```diff
@@ -217,8 +229,69 @@ export function Users() {
             </div>
           </>
         )}
+        <div
+          className={`d-flex align-items-center ${
+            hasSelectedRows ? "" : "invisible"
+          }`}
+        >
+          <Select
+            options={BULK_OPERATIONS.map((op) => ({
+              label: op,
+              value: op,
+            }))}
+            value={
+              bulkOperation
+                ? { label: bulkOperation, value: bulkOperation }
+                : null
+            }
+            onChange={(e) => setBulkOperation(e?.value)}
+            styles={customStyles}
+            isClearable={true}
+            isDisabled={selectedRows?.length === 0}
+            placeholder="Bulk action"
+            aria-label="Bulk action"
+          />
+          &nbsp;
+          {bulkOperation && (
+            <>
+              <button
+                className="ms-2 btn btn-primary"
+                type="button"
+                disabled={!hasSelectedRows || !bulkOperation}
+                onClick={() => {
+                  switch (bulkOperation) {
+                    case "Delete":
+                      askConfirmation(
+                        `Are you sure you want to delete ${
+                          selectedRows.length
+                        } user${selectedRows.length > 1 ? "s" : ""} ?`,
+                        () => {
+                          return Promise.all(
+                            selectedRows.map((row) =>
+                              userDeleteMutation
+                                .mutateAsync(row.username)
+                                .then(() => setBulkOperation(undefined))
+                            )
+                          );
+                        }
+                      );
+                      break;
+                  }
+                }}
+              >
+                {bulkOperation} {selectedRows.length} user
+                {selectedRows.length > 1 ? "s" : ""}
+              </button>
+            </>
+          )}
+        </div>
         <GenericTable
+          selectableRows={user?.admin || isTenantAdmin}
           data={users}
+          onRowSelectionChange={(rows) => {
+            setSelectedRows(rows.filter((r) => checkUserName(r)));
+          }}
+          isRowSelectable={(row) => checkUserName(row)}
```

Comment:
I think checkUserName should be inlined here, and remove from above callback, since it's impossible to select a row that is not selectable.